pkg/agent/util/net_windows.go

```diff
@@ -647,6 +665,18 @@ func HostInterfaceExists(ifaceName string) bool {
 	return true
 }
 
+// InterfaceIPv4DhcpEnabled returns the Ipv4 DHCP status on the specified interface.
+func InterfaceIPv4DhcpEnabled(ifaceName string) (bool, error) {
+	adapter, err := getAdapterInAllCompartmentsByName(ifaceName)
+	if err != nil {
+		return false, err
+	}
+	// IP_ADAPTER_DHCP_ENABLED is defined in the Win32 API document.
+	// https://learn.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_adapter_addresses_lh
+	ipv4Dhcp := (adapter.flags&0x00000004 != 0)
```

Comment:
Use a constant value for `0x00000004` as I saw you also use it in test code.

Reply:
Updated.